macro/src/main/scala/SparkSchemaHelper.scala

```diff
@@ -74,38 +78,53 @@ class SparkSchemaHelperImpl(val c: Context) {
             case x if x =:= typeOf[Long] => seqAny
             case _ => seqRow
           }
-        } else if (isMapOfAny || isMapWithOptionalVal) {
-          mapAny
         } else {
           innerType
         }
 
         val throwError =
-          q"throw cognite.spark.v1.SparkSchemaHelper.badRowError($r, $name, ${rowType.toString}, ${structType.toString})"
+          q"throw cognite.spark.v1.SparkSchemaHelperRuntime.badRowError($r, $name, ${innerType.toString}, ${structType.toString})"
 
         def handleFieldValue(value: Tree) =
-          if (rowType <:< typeOf[Double]) {
+          if (rowType == typeOf[Double]) {
             // do implicit conversion to double
             q"""($value match {
              case x: Double => Some(x)
-             case x: Int => Some(x: Double)
-             case x: Long => Some(x: Double)
+             case x: Int => Some(x.toDouble)
+             case x: Float => Some(x.toDouble)
+             case x: Long => Some(x.toDouble)
              case x: BigDecimal => Some(x.toDouble)
              case x: BigInt => Some(x.toDouble)
              case _ => $throwError
            })"""
-          } else if (rowType <:< typeOf[Long]) {
+          } else if (rowType == typeOf[Float]) {
+            // do implicit conversion to double
```

Comment:
Either the comment or the code is wrong, should probably be `x.toDouble`?

Reply:
Ah, sorry, the comment is lying ...